### PR TITLE
Support Symfony 3

### PR DESCRIPTION
**Pull request by WouterJ**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #317 
| License       | MIT
| Doc PR        | -

Created a new branch, so I don't have to rebase a branch that's used by people already.

### CODE REVIEW

File: Tests/Resources/app/config/cmf_routing.phpcr.yml [hunk 1]
```diff
@@ -3,4 +3,4 @@ cmf_routing:
         persistence:
             phpcr:
                 enabled: true
-                route_basepath: /test/routing
+                route_basepaths: /test/routing
```
Comment (dantleech) [inline Tests/Resources/app/config/cmf_routing.phpcr.yml]: should this be an `[ 'array' ]` ? I know that we probably support scalars, but I think its better to treat a plural as a plural.

Comment (dbu) [inline Tests/Resources/app/config/cmf_routing.phpcr.yml]: not sure. it does not cost us much to convert string to array here for when you have only one path. @WouterJ what do you think is better DX? do we create confusion by allowing single value or is it convenient?

**Comment by dbu** [comment]
hm, why is travis-ci not looking at this?

if this is all it takes, i wonder if we could support symfony 3 with a routing bundle 1.x . should we branch alias to 1.5 and aim to release an 1.5.0 as soon as symfony 3 support is solid?

**Comment by dbu** [comment]
i mean: i don't see any BC breaks in this PR, this is what made me think we could have an 1.x release of this.

**Comment by WouterJ** [comment]
Hmm, indeed, @dbu. However, there is a problem: None of the dev (optional) dependencies support Symfony 3. This means that we can't run the functional and webtests with Symfony 3. It also means that people can't use the admin integration, etc.

If no other CMF bundle supports Symfony 3, does it make sense to support Symfony 3 in this bundle?

**Comment by dbu** [comment]
not sure but i think some people use this without other cmf bundles and
without sonata, so it would be useful to them.

but if we can't run the functional tests, i am not confident to call
things supported. lets be pragmatic and have this go into a 2.0 version
together with the rest then.

**Comment by WouterJ** [comment]
Please note that for Symfony 3 support, people can require `dev-symfony3` as their version constraint btw.

**Comment by dbu** [comment]
can we fix styleci too?

**Comment by WouterJ** [comment]
I now got all tests working on Symfony 3 locally. This requires https://github.com/sonata-project/SonataAdminBundle/pull/3709 and https://github.com/sonata-project/SonataDoctrinePhpcrAdminBundle/pull/367 to be merged first.

**Comment by dbu** [comment]
https://github.com/sonata-project/SonataDoctrinePhpcrAdminBundle/pull/367 is merged, is this now ready?

**Comment by WouterJ** [comment]
I'll try to merge most of the open Sf3 PR this evening

**Comment by WouterJ** [comment]
It took a bit longer, but the road to Symfony 3 support is now officially open :)

**Comment by dbu** [comment]
> It took a bit longer, but the road to Symfony 3 support is now
> officially open :)

woohoo, thanks a lot for your efforts wouter!

**Comment by wachterjohannes** [comment]
@dbu will there be progess for 1.x series? in the future?

**Comment by dbu** [comment]
we try to get the 2.0 versions out of the door. at that point we will not spend time on 1.x anymore, we don't have the resources. as far as i can see, 2.0 will however still support symfony 2.8

**Pull request merged**